src/app/modules/MainScreen/store/actions/index.ts

```diff
@@ -1,27 +1,42 @@
-import { ReturnType } from '@namespace/index';
+import { ReturnType, reduxAsyncAction } from '@namespace/index';
 
 export enum MainScreenTypes {
   INCREMENT = 'INCREMENT',
+  INCREMENT_SUCCESS = 'INCREMENT_SUCCESS',
+  INCREMENT_FAILED = 'INCREMENT_FAILED',
   DECREMENT = 'DECREMENT',
+  DECREMENT_SUCCESS = 'DECREMENT_SUCCESS',
+  DECREMENT_FAILED = 'DECREMENT_FAILED',
   ASYNC_INCREMENT = 'ASYNC_INCREMENT',
+  ASYNC_INCREMENT_SUCCESS = 'ASYNC_INCREMENT_SUCCESS',
+  ASYNC_INCREMENT_FAILED = 'ASYNC_INCREMENT_FAILED',
 }
 
-export const incrementValue = () =>
-  ({
-    type: MainScreenTypes.INCREMENT,
-  } as const);
+export const {
+  execute: incrementValue,
+  success: incrementValueSuccess,
+  failed: incrementValueFailed,
+} = reduxAsyncAction(MainScreenTypes.INCREMENT);
 
-export const decrementValue = () =>
-  ({
-    type: MainScreenTypes.DECREMENT,
-  } as const);
+export const {
+  execute: decrementValue,
+  success: decrementValueSuccess,
+  failed: decrementValueFailed,
+} = reduxAsyncAction(MainScreenTypes.DECREMENT);
 
-export const incrementAsyncValue = () =>
-  ({
-    type: MainScreenTypes.ASYNC_INCREMENT,
-  } as const);
+export const {
+  execute: asyncIncrementValue,
+  success: asyncIncrementValueSuccess,
+  failed: asyncIncrementValueFailed,
+} = reduxAsyncAction(MainScreenTypes.ASYNC_INCREMENT);
 
 export type MainScreenActions =
   | ReturnType<typeof incrementValue>
+  | ReturnType<typeof incrementValueSuccess>
```

Comment:
@eliotmess Please focus on recreatimg fucntionality we have in standard React app and remove this dummy increase/decrease functionality :)